### ISSUE TITLE
Candidacy process interrupted, editor caused CPU load, transaction not picked up

### ISSUE THREAD

**Issue by traumschule** [Joystream/pioneer#3291]
Route: https://dao.joystream.org/#/election, last step of `Announce Candidacy` modal

TLDR Users wan't to use keyboard navigation and click to position the cursor without the editor changing cursor position.

Reported [here](https://discord.com/channels/811216481340751934/812344874099277824/989887189842296842) and [here](https://discord.com/channels/811216481340751934/812343711870091285/989895511727276032).

![](https://cdn.discordapp.com/attachments/812344874099277824/989887189095678022/pioneer-cpu.png)

The candidacy editor does some tricks to position cursor (or it may be a side-effect of `onChange` updating content), i suggest to remove that / time it better (like after 500ms user inactivity or similar). Reasons and observations:

> Editing text for candidacy announcement .. Some really weird cursor jumping going on.
> seeing a regression when announcing candidacy. It fails to pick up the transaction again. Also CPU load is >80% = not ok. Also fails to redraw screen.
> tried to reproduce as well, did not happen yet 
> It could be the editor that does some cursor tricks similar to what was fixed (i think) for the forum. Maybe best to remove that code.
Like i was unable to mark two words either with the mouse or Shift + Arrow to set a link because it was repositioning the cursor all the time.
> CPU load was massive which could be the reason that the transaction wasn't picked up in time. No idea what triggered it but something was stuck in a loop.

![](https://cdn.discordapp.com/attachments/812343711870091285/989895510976528444/pioneer-hangs.png)

related:
- https://github.com/Joystream/pioneer/issues/2278
- https://github.com/Joystream/pioneer/pull/1600
- https://github.com/Joystream/pioneer/blob/825d327e4b49d7ad056b0e73b4004152cf6f04e1/packages/ui/src/common/components/CKEditor/CKEditor.tsx#L134

**Comment by dmtrjsg** [comment]
We are not sure what can be done here, so pls let me know if this repeats and we will re-open. @traumschule

**Comment by traumschule** [comment]
It needs a performance audit, for example with dev tools.

**Comment by dmtrjsg** [comment]
@traumschule ok re-opened, added to community pipe, pls feel free to take this one on at some point. But pls bear in mind that issues marked with label:mainnet are of highest prio. 

cc @singulart

**Comment by ivanturlakov** [comment]
✅ Tested on https://dao-git-carthage-joystream.vercel.app/#/election
wss://44.201.112.198.nip.io/ws-rpc
<img width="326" alt="Снимок экрана 2022-11-02 в 14 29 48" src="https://user-images.githubusercontent.com/46903215/199489894-f108601d-392f-439e-aed3-1f222e7c0726.png">
Chrome 107.0.5304.87, (x86_64)

While filling out the application, I tried to load the editor as much as possible and keep the modal open for a long time. Sometimes, especially when I inserted large chunks of text into the editor, the CPU load jumped up to 90 percent, but then quickly returned to normal. In general, I did not encounter the problems described above, the process of editing and submission was successful.

https://user-images.githubusercontent.com/46903215/199493957-f26c83db-77b3-4098-b960-e3a33b3e1a15.mov

**Comment by traumschule** [comment]
Unfortunately i got some to back this up again.

I just filled out my candidacy and it felt like the tab would crash anytime. At first everything was fine filling bullet points, adding a description, going back and forth but i think what "broke" it (ie caused some kind of loop and continues load) was the editor as assumed before.

Specifically links and marking text via cursor (shift + arrows) or cursor selection fails because the editor does something to reset it and reposition the cursor (i know this was mentioned before).
I noticed the edit link feature is broken as in after editing it keeps the cursor at the position and wants me to keep editing the link until i click it, then the pen and abort to "achieve" free editing again.

Finally was able to start the signing process, the extension tab opened, password entered and confirmed. Going back to the tab it looked less comforting and i had to assume everything was lost (knowing it does not backup to browser store after editing).

After that something sad happened and i don't know why:
> 2022-11-08 10:26:56        RPC-CORE: submitAndWatchExtrinsic(extrinsic: Extrinsic): ExtrinsicStatus:: 1010: Invalid Transaction: Transaction has a bad signature
> logger.cjs:63 2022-11-08 10:26:56             DRR: 1010: Invalid Transaction: Transaction has a bad signature
> logger.cjs:63 2022-11-08 10:26:56        RPC-CORE: submitAndWatchExtrinsic(extrinsic: Extrinsic): ExtrinsicStatus:: 1010: Invalid Transaction: Transaction has a bad signature
﻿
Now the modal is in that state with inactive buttons and no content. Probably got to scrape my text from RAM or a browser core dump.

</SadUserstory>

After click to sign before activating extension
![candidacy-load](https://user-images.githubusercontent.com/31551045/200545370-f3775527-c83a-4973-ab2e-f0e88dabe7e6.png)

Long waiting for page rebuild/-render

![looks-broken](https://user-images.githubusercontent.com/31551045/200545195-4115e167-8f76-4f9c-86dc-bd7ddadb617d.png)

After signing
![after-signing](https://user-images.githubusercontent.com/31551045/200545219-451e1a89-80c0-4b24-806e-31e31dbbeb55.png)

Final state
![page-rebuilt](https://user-images.githubusercontent.com/31551045/200545285-49dc034b-7e09-48db-87ac-7f1528b72684.png)

=> allow user to retry signing after failed attempt

**Comment by traumschule** [comment]
not reproducible on dev, closing
